### PR TITLE
Use artifactory-papi:7.12.5 CLM-22487

### PR DESCRIPTION
**Pull request by vladt**
https://issues.sonatype.org/browse/CLM-22487
https://jenkins.ci.sonatype.dev/job/insight/job/firewall-for-artifactory/job/feature-snapshots/job/CLM-22487_UseArtifactoryPAPI7/

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 make(
     deployBranch: 'main',
-    javaVersion: 'OpenJDK 8',
+    javaVersion: 'OpenJDK 11',
```
Comment (vladt) [inline Jenkinsfile:18]: Artifactory 7.x requires Java 11

File: Jenkinsfile [hunk 2]
```diff
@@ -27,24 +27,24 @@ make(
...
-            }
-            finally {
-              archiveArtifacts(artifacts: '**/target/*-reports/**,**/module.xml')
-              collectTestResults(['**/target/*-reports/*.xml'])
-              deleteDir()
-            }
-          }
-        }
-      }]})
+      //parallel(['OpenJDK 11'].collectEntries { javaVersion -> [javaVersion, {
```
Comment (vladt) [inline Jenkinsfile:30]: Leaving it here as we may need it again in the future

File: firewall/src/main/java/com/sonatype/iq/artifactory/audit/AuditExecutor.groovy [hunk 3]
```diff
@@ -64,7 +63,7 @@ class AuditExecutor
     futuresByRepo.get(repo.repoKey) << future
     addCallback(future, new FutureCallback<AuditResult>() {
       @Override
-      void onSuccess(@NullableDecl final AuditResult result) {
+      void onSuccess(final AuditResult result) {
```
Comment (vladt) [inline firewall/src/main/java/com/sonatype/iq/artifactory/audit/AuditExecutor.groovy:66]: The `@NullableDecl` annotation comes from a transitive dependency that's not provided anymore.

File: pom.xml [hunk 4]
```diff
@@ -39,15 +39,15 @@
...
 
   <properties>
-    <maven.compiler.source>1.8</maven.compiler.source>
-    <maven.compiler.target>1.8</maven.compiler.target>
+    <maven.compiler.source>11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
     <main.basedir>${project.basedir}</main.basedir>
 
-    <artifactory.version>6.6.5</artifactory.version>
+    <artifactory.version>7.12.5</artifactory.version>
```
Comment (vladt) [inline pom.xml:46]: This is the newest version I could find.

File: pom.xml [hunk 5]
```diff
@@ -39,15 +39,15 @@
...
     <!-- Provided by artifactory-papi -->
-    <guava.version>23.6.1-jre</guava.version>
+    <guava.version>28.2-jre</guava.version>
```
Comment (vladt) [inline pom.xml:48]: Match version used by Artifactory

File: pom.xml [hunk 6]
```diff
@@ -39,15 +39,15 @@
...
     <!-- Provided by artifactory-papi -->
-    <xstream.version>1.4.10</xstream.version>
+    <xstream.version>1.4.11.1</xstream.version>
```
Comment (vladt) [inline pom.xml:50]: Match version used by Artifactory

**Review by DanielRuaMadrid — APPROVED** [review]
+1

**Review by laas29 — APPROVED** [review]

**Pull request merged**